src/System.Private.ServiceModel/tools/PRService/sync.ashx

```diff
@@ -0,0 +1,227 @@
+<%@ WebHandler Language="C#" Class="PullRequestHandler" %>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+// This script assumes that the repo is set up as follows: 
+
+//[remote "origin"]
+//	url = https://github.com/dotnet/wcf
+//	fetch = +refs/heads/*:refs/remotes/origin/*
+//	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files (x86)\Git\bin\git.exe";
+    private const string _gitRepoPath = @"N:\git\wcf-test";
+
+    private const int _gitSanityExecutionTimeoutMilliseconds = 1000;
+    private const int _gitExecutionTimeoutMilliseconds = 60000;
```

Comment:
I ran into an issue with a 60s timeout, actually, turns out that a sync can take a long, long time, either due to slow connection or necessity to pull a lot of files. I kind of like having the timeout here so that we get the execution result when an error happens. 

Based on your feedback for 110s ASP.NET request timeout, I upped the one here to 100s.